dbms/src/Interpreters/Context.h

```diff
@@ -211,8 +212,10 @@ class Context
     std::unique_ptr<DDLGuard> getDDLGuardIfTableDoesntExist(const String & database, const String & table, const String & message) const;
 
     String getCurrentDatabase() const;
+    String getCurrentTable() const;
     String getCurrentQueryId() const;
     void setCurrentDatabase(const String & name);
+    void setCurrentTable(const String & database, const String & table);
```

Comment:
It's a little bit misleading because current database is a database which is used by user. But current table here is a table is used for insert query.

Also this function changes current database. It seems that if we will say `insert into new_database.new_table`, out current database will be `new_database`

Reply:
It changes query context. All the changes would be lost when a new query starts. It would be restored to session context values, that are "default" and <none>. At least they should.